client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
+		const post = getSitePost( getState(), action.siteId, action.postId );
+
+		let message;
+		if ( post ) {
+			message = translate( 'An error occurred while deleting "%s"', {
+				args: [ truncate( post.title, { length: 24 } ) ]
+			} );
+		} else {
+			message = translate( 'An error occurred while deleting the post' );
+		}
+
+		dispatch( errorNotice( message ) );
+	},
+	[ POST_DELETE_SUCCESS ]: ( action, dispatch ) => {
+		dispatch( successNotice( translate( 'Post successfully deleted' ) ) );
+	},
+	[ POST_SAVE_SUCCESS ]: ( action, dispatch ) => {
+		let text;
+		switch ( action.post.status ) {
+			case 'trash':
+				text = translate( 'Post successfully moved to trash' );
+				break;
+
+			case 'publish':
+				text = translate( 'Post successfully published' );
+				break;
+		}
+
+		if ( text ) {
+			dispatch( successNotice( text ) );
+		}
+	}
```

Comment:
these aren't really single-fine functions here for the most part, and they have their own independent behaviors. couldn't we define these functions independently and just map them into the object? I realize this would involve some duplicate code, but I find it easier to scan through and test

```js
export const showMessage = message => ( action, dispatch ) => dispatch( successNotice( message ) );
export const postWasSaved = ( action, dispatch ) => { let text; … };

const observers = [
	…,
	[ POST_DELETE_SUCCESS ]: showMessage( translate( 'Post successfully deleted' ) ),
	[ POST_SAVE_SUCCESS ]: postWasSaved,
	…
];
```

Now to test these things we don't have to import a big object of nothing but functions, rather we can just import those functions directly. Why would a _unit test_ even need to know that the handling code is inside some bigger data structure?